### ISSUE TITLE
Anchor point broken

### ISSUE THREAD

**Issue by mikeevmm** [HaxeFlixel/flixel#1920]
- **Flixel version:** 4.1.1
- **OpenFL version:** 3.6.1
- **Lime version:** 2.9.1
- **Affected targets:** Native. Flash

________________________________

**Code snippet reproducing the issue**:

```haxe
package;

import flixel.FlxState;
import flixel.FlxSprite;

class PlayState extends FlxState
{
	override public function create():Void
	{
		var sprite:FlxSprite = new FlxSprite();
		sprite.loadGraphic("anyGraphic.png");
		sprite.scale.set(FlxG.width / sprite.width, FlxG.height / sprite.height);
		sprite.setPosition(FlxG.camera.scroll.x, FlxG.camera.scroll.y);
		add(sprite);
	}
}
```

________________________________

**Observed behavior:**
![image](https://cloud.githubusercontent.com/assets/7550678/18070004/e18b65d0-6e42-11e6-963f-1e68d0ffebcd.png)
**Expected behavior:**
![image](https://cloud.githubusercontent.com/assets/7550678/18070311/e638a870-6e44-11e6-92ff-41489fd29aa7.png)

**Comment by IBwWG** [comment]
I think you need to call `sprite.updateHitbox()`

**Comment by Gama11** [comment]
@IBwWG That wouldn't make any sense, he's not relying on `width` / `height` for anything after scaling. After loading the image, `width == frameWidth` and `height == frameHeight`.

**Comment by IBwWG** [comment]
Oh sorry...I thought I remembered a very similar situation where that's what I'd needed to do (after changing the scale and position.)  I guess that's only if you're centering and need to access the width value, from how you explained just now?

**Comment by Gama11** [comment]
You could just use the graphic width (`frameWidth`) instead of the hitbox width (`width`) in that situation.
Admittedly, the naming is a bit confusing and the difference not always clear to users.

**Comment by Gama11** [comment]
Actually, forget that, `frameWidth` / `Height` in unscaled...

**Comment by IBwWG** [comment]
Uhoh, when even *you* can't remember... ;)

**Comment by mikeevmm** [comment]
`updateHitbox` offsets the sprite even more.

Oddly enough, using  `width` to center the sprite works correctly.

**Comment by Beeblerox** [comment]
actually it's not broken. It behaves just like in original flixel. But this behavior isn't quite logical.
But in your case (only if you're not rotating sprite) it could be fixed by adding one line:
```
sprite.origin.set(0, 0);
```

I'm planning to change anchor/origin point behavior, but it will be major breaking change, so everyone have to wait till next major release (5.0?)

**Comment by Beeblerox** [comment]
i would be glad to help if you have any questions/problems with anchor points in flixel.
Just ask them here. Ok?

**Comment by Gama11** [comment]
I think this can be closed then, we have #923 to track the `origin` change you want to make.
